app/index.coffee

```diff
@@ -105,6 +107,11 @@ Api.proxy.el().one 'load', ->
       app.topBar.loginForm.signInButton.click()
       app.topBar.loginDialog.reattach()
 
+    $(window).bind('beforeunload', (e) ->
+        AnalyticsLogger.logEvent User.current?.zooniverse_id,Subject.current?.zooniverseId,'leave',''
+        event.preventDefault()
```

Comment:
played with this a bit more. the event.preventDefault() is definitely required - anything else and we get a nasty popup.